fields.go

```diff
@@ -21,21 +21,38 @@ type EncodedFields []EncodedField
 
 // PrependUnique encoded field if the key is not already set
 func (f EncodedFields) PrependUnique(fields EncodedFields) EncodedFields {
+	var res EncodedFields
+	var offset int
 	for ix := len(fields) - 1; ix >= 0; ix-- {
 		field := fields[ix]
-
+		key := field.Key()
```

Comment:
This looks like an unnecessary assignment

Reply:
It can be moved down to line 32, but it is used in the search loop.

Reply:
yeah - it's assigned to `key` and only used in the one place i mean, and i don't think it aids the readability over just using `if v.Key() == field.Key()` directly.

Reply:
It is given it's own variable to not have to call Key method on every iteration of the search loop.